calico_kubernetes/calico_kubernetes.py

```diff
@@ -505,20 +504,9 @@ def _get_pod_ports(self, pod):
 
     def _get_pod_config(self):
         """Get the list of pods from the Kube API server."""
-        pods = self._get_api_path('pods')
-        logger.debug('Got pods %s', pods)
-
-        for pod in pods:
-            logger.debug('Processing pod %s', pod)
-            ns = pod['metadata']['namespace'].replace('/', '_')
-            name = pod['metadata']['name'].replace('/', '_')
-            if ns == self.namespace and name == self.pod_name:
-                this_pod = pod
-                break
-        else:
-            raise KeyError('Pod not found: ' + self.pod_name)
-        logger.debug('Got pod data %s', this_pod)
-        return this_pod
+        pod = self._get_api_path('namespaces/%s/pods/%s' % (self.namespace, self.pod_name))
```

Comment:
I like this change.

However, I think `_get_api_path` needs some thinking.  Currently, we only use it to get the `pod` API object.  However, its name suggests something more general purpose, and the docstring suggests it only returns a list of objects :)

I think keeping the name and updating the docstring to describe what is actually returned probably makes the most sense.